server/jobs/bsve_search_worker.py

```diff
@@ -88,12 +86,42 @@ def run(job):
         jsonRow = jsonArrayHead(humanFilepath, limit=1)[0]
         minerva_metadata['json_row'] = jsonRow
 
+        # Generate the geojson for this dataset and set
+        # dataset_type = geojson
+
+        geojsonFilename = 'search.geojson'
+        geojsonFilepath = os.path.join(tmpdir, geojsonFilename)
+
+        mapping = {
+            "dateKeypath": "",
+            "latitudeKeypath": "data.Latitude",
+            "longitudeKeypath": "data.Longitude"
+        }
+
+        geojsonMapper = GeoJsonMapper(objConverter=None, mapping=mapping)
+        objects = jsonObjectReader(humanFilepath)
+        geojsonMapper.mapToJsonFile(tmpdir, objects, geojsonFilepath)
+
+        client.uploadFileToItem(datasetId, geojsonFilepath)
         shutil.rmtree(tmpdir)
 
-        metadata['minerva'] = minerva_metadata
-        # TODO only works locally
-        item_model.setMetadata(dataset, metadata)
-        # TODO only works locally
+        minerva_metadata['mapper'] = mapping
+        minerva_metadata['dataset_type'] = 'geojson'
+
+        existing = file_model.findOne({
+            'itemId': dataset['_id'],
+            'name': geojsonFilename
+        })
+        if existing:
+            minerva_metadata['geojson_file'] = {
+                '_id': existing['_id'],
+                'name': geojsonFilename
+            }
+        else:
+            raise (Exception('Cannot find file %s in dataset %s' %
+                   (geojsonFilename, datasetId)))
+
+        mM(dataset, minerva_metadata)
         job_model.updateJob(job, status=JobStatus.SUCCESS)
```

Comment:
I know calling it just ```metadata``` here would be ambiguous, but can we find some way to rename ```mM``` server-side too?